hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/cancelairdrops/TokenCancelAirdropDecoderTest.java

```diff
@@ -0,0 +1,327 @@
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L)
+        })));
+        given(attempt.inputBytes()).willReturn(encoded.toArrayUnsafe());
+
+        assertThrows(HandleException.class, () -> subject.decodeCancelAirdrop(attempt));
+    }
+
+    @Test
+    void failsIfTokenIsNull() {
+        // given:
+        given(attempt.configuration()).willReturn(configuration);
+        given(attempt.enhancement()).willReturn(enhancement);
+        given(enhancement.nativeOperations()).willReturn(nativeOperations);
+        given(configuration.getConfigData(TokensConfig.class)).willReturn(tokensConfig);
+        given(tokensConfig.maxAllowedPendingAirdropsToCancel()).willReturn(10);
+        given(nativeOperations.getToken(FUNGIBLE_TOKEN_ID.tokenNum())).willReturn(null);
+        given(addressIdConverter.convert(asHeadlongAddress(SENDER_ID.accountNum())))
+                .willReturn(SENDER_ID);
+        given(addressIdConverter.convert(OWNER_ACCOUNT_AS_ADDRESS)).willReturn(OWNER_ID);
+
+        final var encoded = Bytes.wrapByteBuffer(CANCEL_AIRDROP.encodeCall(Tuple.singleton(new Tuple[] {
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L)
+        })));
+        given(attempt.inputBytes()).willReturn(encoded.toArrayUnsafe());
+
+        assertThrows(
+                HandleException.class,
+                () -> subject.decodeCancelAirdrop(attempt),
+                PENDING_AIRDROP_ID_LIST_TOO_LONG.protoName());
```

Comment:
Should this be `PENDING_AIRDROP_ID_LIST_TOO_LONG` -> `INVALID_TOKEN_ID` and if so why doesn't this test fail?

And the answer why it doesn't fail is:  the `message` argument to `assertThrows` is _not_ a matcher against the message of the exception, instead it is the message for Junit to report for the test failure.  (I didn't realize this until I debugged into it.)

Suggest using instead AssertJ's `assertThat(thrown).isInstanceOf(...).hasMessage(...)` or, to match this style here (which I prefer)

```
assertThatExceptionOfType(HandleException.class)
   .isThrownBy(() -> ...some lambda body here...)
   .withMessage("...");
```

Same problem appears in following tests.

Reply:
Adjusted